pkg/build/api/v1/conversion_test.go

```diff
@@ -84,7 +84,8 @@ func TestBuildConfigConversion(t *testing.T) {
 				},
 				Triggers: []older.BuildTriggerPolicy{
 					{
-						Type: older.ImageChangeBuildTriggerTypeDeprecated,
+						Type:        older.ImageChangeBuildTriggerTypeDeprecated,
+						ImageChange: &older.ImageChangeTrigger{},
```

Comment:
Conversion will barf on a nil ImageChange field if the trigger type is ImageChangeTriggerType.

I had convinced myself that was ok because validation would enforce that it not be nil but now i think i realize that's not good enough since conversion will happen before validation.  So i guess i'll make the conversion resilient.